src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -92,17 +97,17 @@ public interface UserRepository extends JpaRepository<User, Long>, JpaSpecificat
     @EntityGraph(type = LOAD, attributePaths = { "groups", "authorities", "guidedTourSettings" })
     Optional<User> findOneWithGroupsAuthoritiesAndGuidedTourSettingsByLogin(String login);
 
-    @Query("select count(*) from User user where :#{#groupName} member of user.groups")
+    @Query("select count(*) from User user where user.isDeleted = false and :#{#groupName} member of user.groups")
```

Comment:
Maybe it would be easier to create a database view `NonDeletedUsers` or `ActiveUsers`